src/java.base/share/classes/sun/security/ssl/SSLAlgorithmConstraints.java

```diff
@@ -63,42 +63,58 @@ final class SSLAlgorithmConstraints implements AlgorithmConstraints {
     static final AlgorithmConstraints DEFAULT_SSL_ONLY =
                         new SSLAlgorithmConstraints((SSLSocket)null, false);
 
-    SSLAlgorithmConstraints(AlgorithmConstraints userSpecifiedConstraints) {
+    private SSLAlgorithmConstraints(AlgorithmConstraints userSpecifiedConstraints) {
         this.userSpecifiedConstraints = userSpecifiedConstraints;
         this.peerSpecifiedConstraints = null;
         this.enabledX509DisabledAlgConstraints = true;
     }
 
+    static AlgorithmConstraints wrap(AlgorithmConstraints userSpecifiedConstraints) {
+        if (userSpecifiedConstraints == DEFAULT) {
```

Comment:
Maybe, DEFAULT could be returned for null userSpecifiedConstraints.

```
-        if (userSpecifiedConstraints == DEFAULT) {
+        if (userSpecifiedConstraints == null &&
+                 userSpecifiedConstraints== DEFAULT) { 
```

Reply:
@XueleiFan did you mean `||` (not `&&`) ?

Reply:
> @XueleiFan did you mean `||` (not `&&`) ?

Thank you @dfuch.  Yes, it should be "||".

Reply:
as of today, this method is never called with a `null` argument (`SSLConfiguration#userSpecifiedAlgorithmConstraints` is initialized to `DEFAULT` and cannot be reset to `null`), but I can add a null check for future-proofing.

Reply:
I know.  But if the null condition is not added, a code reader may have to search for its usage and make sure null is not passed.  If the usages are in the same class, I may not add the checking.  Otherwise, an additional checking might save time in the future.